Makefile

```diff
@@ -81,7 +81,12 @@ upload:
 	svn import "$(ARCHIVE_DIR)/$(EXPORT_WIN32)" "$(REPO_TAGS)/archives/$(EXPORT_WIN32)" \
 		-m "added released win32 installer for version $(VERSION)"
 
-test: check-style
+test: check-style pytest
+
+pytest:
+	for TEST in $$(find tests/ -name '*.py'); do \
+	    python -m pytest -v $$TEST; \
+	done
```

Comment:
pytest has builtin logic for traversing down from the directory it receives as an argument.  For each directory in the target recurse into that directory and collect all of the "tests" within files named:  `test_*.py`   (also perhaps other files that include the word "test" in their name).
So, we don't need to manually feed `pytest` each test target.   We should be able to simply say:

`pytest -v ${PROJECTROOT}`

Reply:
Oh...   huh...   pytest didn't pick up the "tests" tests...
...

Reply:
Ahh...   let's `mv tests/Geometry/Polygon.py tests/Geometry/test_Polygon.py`

Or maybe `Test_Polygon.py` testing...

Reply:
@SebKuzminsky your choice, I think it's sufficient to start the module name with "test_" and I _think_ it's case insensitive.